app/controllers/locks_controller.rb

```diff
@@ -1,32 +1,51 @@
 class LocksController < ApplicationController
-  before_action :authorize_deployer!
+  include ProjectLevelAuthorization
+
+  before_action unless: :for_global_lock? do
+    find_project
+    authorize_project_deployer!
+  end
+
   before_action :authorize_admin!, if: :for_global_lock?
 
   def create
     attributes = params.require(:lock).
-      permit(:description, :stage_id, :warning).
-      merge(user: current_user)
+        permit(:description, :stage_id, :warning).
+        merge(user: current_user)
     Lock.create!(attributes)
-    redirect_to :back, notice: "Locked"
+    redirect_to :back, notice: 'Locked'
   end
 
   def destroy
     lock.try(:soft_delete)
-    redirect_to :back, notice: "Unlocked"
+    redirect_to :back, notice: 'Unlocked'
   end
 
   protected
 
   def for_global_lock?
     case action_name
-    when "create" then !params[:lock].try(:[], :stage_id)
-    when "destroy" then !lock.stage_id
-    else
-      raise "Unsupported action"
+      when 'create' then
+        !params[:lock].try(:[], :stage_id) || params[:lock].try(:[], :stage_id).try(:empty?)
```

Comment:
This one was hard to remember why did I changed it. :-P
But yes, I noticed that `stage_id` is always present in the parameters, and when to provided, it comes as an empty string.

I was not aware of the `presence` method. Looks much better, thanks. I'll check it out.